examples/switch_client_async_nats.rs

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: 2021 Open Energy Solutions Inc
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::StreamExt;
+use log::info;
+use openfmb::{
+    encoding::ProtobufEncoding,
+    prelude::Publisher,
+    topic::{self, ProfileTopic},
+};
+use openfmb_messages::{switchmodule::SwitchDiscreteControlProfile, Module, Profile};
+use openfmb_messages_ext::SwitchControlExt;
+use std::{env, str::FromStr};
+
+fn topic(profile: topic::Profile, mrid: &uuid::Uuid) -> ProfileTopic {
+    ProfileTopic::new(Module::SwitchModule, profile, mrid.clone())
+}
+
+#[tokio::main]
+pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    pretty_env_logger::init();
+    let mrid = uuid::Uuid::from_str(
+        &env::var("SWITCH_MRID").unwrap_or("ec9dc3dd-d329-4ec6-9149-762b033f0be8".to_string()),
+    )?;
+
+    let nats_url = env::var("NATS_URL").unwrap_or("nats://192.168.86.30:4222".to_string());
+    let nc = async_nats::connect(&nats_url).await?;
```

Comment:
Would be nice to have an example of connection that handles error and callback events